pulp_service/pulp_service/app/tasks/package_scan.py

```diff
@@ -0,0 +1,105 @@
+import aiohttp
+import logging
+import json
+
+from asgiref.sync import sync_to_async
+
+from pulpcore.app.models import Content, RepositoryVersion
+from pulp_rpm.app.models.package import Package as RPMPackage
+from pulp_npm.app.models import Package as NPMPackage
+
+from pulp_service.app.constants import (
+    RH_REPO_TO_CPE_URL,
+    PKG_ECOSYSTEM,
+    OSV_QUERY_URL,
+)
+from pulp_service.app.models import ArtifactVulnerability
+
+_logger = logging.getLogger(__name__)
+cache_rh_cpe = {}
+
+
+async def check_content(repo_version_pk):
+    """
+    Get the list of contents from reop_version, build a package_list and makes an API request to
+    osv.dev using this package_list
+    """
+    contents = await sync_to_async(_get_content_from_repo)(repo_version_pk)
+    osv_packages = await sync_to_async(_define_osv_package_list)(contents)
+    if not osv_packages:
+        return
+    await _scan_packages(osv_packages)
+
+
+async def _scan_packages(packages):
+    """
+    Makes a request to the osv.dev API and store the results in ArtifactVulnerability model
+    """
+    async with aiohttp.ClientSession() as session:
+        for package in packages:
+            osv_data = json.dumps(package["osv_data"])
+            async with session.post(url=OSV_QUERY_URL, data=osv_data) as response:
+                response_body = await response.text()
+                json_body = json.loads(response_body)
+                if json_body.get("vulns"):
+                    await sync_to_async(ArtifactVulnerability.objects.get_or_create)(
```

Comment:
I envisioned a single report per scan. The JSON would contain a key for each package. and the value for that key would be the scan results for that package.